File: samples/php/chstatus_inq.php
<?php
	/*
	 * Inquire all server connection channel statuses for queuemanager PIGEON.
	 * MQWeb runs on localhost and is listening on port 8081. 
	 */
	$url = "http://localhost:8081/api/chstatus/inquire/PIGEON?ChannelType=Server-connection";
	
	$curl = curl_init();
	curl_setopt($curl, CURLOPT_URL, $url);
	curl_setopt($curl, CURLOPT_RETURNTRANSFER, 1);
	
	if ( ($response = curl_exec($curl)) === false )	{
		$err = curl_error($curl);
		echo "An HTTP error occurred while inquiring channel status information: $err\n";
	}
	else {
		$json = json_decode($response);
		if ( isset($json->error) ) {
			echo "An MQ error occurred while inquiring channel statuses.\n";
			echo "Reason Code: {$json->error->reason->code} - {$json->error->reason->desc}\n";
		}
		else {
			if ( isset($json->data) && count($json->data) > 0 ) {
				foreach($json->data as $status)
				{
					echo $status->ChannelName->value;
					echo " : ";
					echo $status->ChannelStatus->display;
					echo "\n";
				}
			}
			else
			{
				echo "No statuses found\n";
			}
		}
	}

File: samples/python/chstatus_inq.py
'''
 Inquire all server connection channel statuses for a queuemanager.
 MQWeb runs on localhost and is listening on port 8081.
'''
import sys
import json
import httplib
import socket

if len(sys.argv) < 2 :
	print 'Please pass me the name of a queuemanager as argument'
	sys.exit(1)

url = '/api/chstatus/inquire/' + sys.argv[1] + '?ChannelType=Server-connection'

try:
	conn = httplib.HTTPConnection('localhost', 8081)
	conn.request('GET', url)
	res = conn.getresponse()
	result = json.loads(res.read())

	if 'error' in result:
		print ('Received a WebSphere MQ error: ' +	
			str(result['error']['reason']['code'])
		)
	else:
		if len(result['data']) == 0:
			print "No channelstatus found"
		else:
			for data in result['data']:
				print (data['ChannelName']['value'] + ': ' 
					+ data['ChannelStatus']['text']
				)
except httplib.HTTPException as e:
	print ('An HTTP error occurred while inquiring channel statuses: ' +
		e.errno + e.strerror
	)
except socket.error as e:
	print e.strerror
	print 'Is the MQWeb daemon running?'
